### PR TITLE
Accept io.Data as input.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: tests/ar_test.toit [hunk 1]
```diff
@@ -34,6 +34,7 @@ write-ar file-mapping/Map --add-with-ar-file/bool=false:
   writer := io.Buffer
   ar-writer := ArWriter writer
   file-mapping.do: |name content|
+    if content is string: content = content.to-byte-array
```
Comment (kasperl) [inline tests/ar_test.toit]: Why is this necessary?

Comment (floitsch) [reply tests/ar_test.toit]: good catch.
should have in in the next line.
done.

File: tests/ar_test.toit [hunk 2]
```diff
@@ -35,6 +35,8 @@ write-ar file-mapping/Map --add-with-ar-file/bool=false:
   ar-writer := ArWriter writer
   file-mapping.do: |name content|
     if add-with-ar-file:
+      // The `ArFile` only takes ByteArrays.
```
Comment (kasperl) [inline tests/ar_test.toit]:
```suggestion
      // The `ArFile` only takes byte arrays.
```

**Review by kasperl — APPROVED** [review]
Looks good to me.
